src/summarizedexperiment/utils/_slicer.py

```diff
@@ -9,34 +9,31 @@
 
 
 def get_indexes_from_names(
-    _source: Sequence[Any], _target: Sequence[Any]
-) -> np.ndarray:
-    """Get the indexes in source where values in target are found.
+    source: Sequence[Any], target: Sequence[Any]
+) -> Sequence[int]:
+    """Return the index of the first occurrence of each value in `source`.
 
     Args:
-        _source (Sequence[Any]): the source index names.
-        _target (Sequence[Any]): the target index names.
+        source (Sequence[Any]): the list in which to search for the values.
+        target (Sequence[Any]): the list of values to find indices for.
 
     Raises:
-        ValueError: If passed index names do not exist in the samples.
-        ValueError: If passed index names is not a collection.
+        ValueError: if any value in `target` is not found in `source`.
 
     Returns:
-        np.ndarray: Integers from 0 to n - 1 indicating that the index at
-            these positions matches the corresponding target values. Missing
-            values in the target are marked by -1.
+        Sequence[int]: the indexes of the first occurrence of each value in
+            `target` within `source`.
     """
-    try:
-        source = pd.Index(_source)
-        target = pd.Index(_target)
-    except TypeError as exception:
-        raise TypeError(f"{_target} is not a collection") from exception
-
-    missing_names = target.difference(source)
-    if not missing_names.empty:
-        raise ValueError("invalid index name(s): " + ", ".join(missing_names))
-
-    return source.get_indexer(target).tolist()
+    missing_names = [value for value in target if value not in source]
+    if len(missing_names) > 0:
+        raise ValueError("invalid index names(s): " + ", ".join(missing_names))
```

Comment:
using sets is much better than iteration

`missing_names = set(source).difference(target)`

Reply:
this works
`missing_names = set(target).difference(source)`